server/src/main/java/org/apache/druid/server/coordinator/DruidCoordinator.java

```diff
@@ -624,6 +630,27 @@ CompactSegments initializeCompactSegmentsDuty(CompactionSegmentSearchPolicy comp
     }
   }
 
+  @Nullable
+  @VisibleForTesting
+  KillUnusedSegments initializeKillUnusedSegmentsDuty()
+  {
+    List<KillUnusedSegments> killUnusedSegmentsDutyFromCustomGroups = getKillUnusedSegmentsDutyFromCustomGroups();
+    if (!config.getAutoKillEnabled()) {
+      return null;
+    }
+    if (killUnusedSegmentsDutyFromCustomGroups.isEmpty()) {
+      return new KillUnusedSegments(segmentsMetadataManager, overlordClient, config);
+    } else {
+      if (killUnusedSegmentsDutyFromCustomGroups.size() > 1) {
+        log.warn(
```

Comment:
Can you throw an exception here instead. This is a misconfiguration, and it would be better to fail startup so the operator can fix the config issue.

In the exception message can you add the names of the custom groups where the duty is configured to make it easy for the operator to fix the config issue.

Reply:
done.